protocol/x/clob/keeper/orders.go

```diff
@@ -241,6 +244,97 @@ func (k Keeper) PlaceShortTermOrder(
 	return orderSizeOptimisticallyFilledFromMatchingQuantums, orderStatus, err
 }
 
+// DeliverMsgCancelOrder delivers a MsgCancelOrder.
+func (k Keeper) DeliverMsgCancelOrder(
```

Comment:
nit: let's maybe call this `HandleMsgCancelOrder`? can also add a `lib.AssertDeliverTxMode(ctx)` or sometihng